src/sciline/pipeline.py

```diff
@@ -406,12 +418,12 @@ def set_param_table(self, params: ParamTable) -> None:
             for index, label in zip(params.index, values):
                 self._set_provider(
                     Item((Label(tp=params.row_dim, index=index),), param_name),
-                    _provider_for_constant(label),
+                    Provider.table(label),
                 )
         for index, label in zip(params.index, params.index):
             self._set_provider(
                 Item((Label(tp=params.row_dim, index=index),), params.row_dim),
```

Comment:
Not necessary, but could this clumsy key setup be absorbed into the `Provider` class, similar to the `function` case? `Provider.table` would obviously need more arguments.

Reply:
Would you store the key in the provider? This would duplicate the key as we still need it in the graph / pipeline.